cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -192,15 +192,18 @@ func (o *ScaleUpOrchestrator) ScaleUp(
 		// one should be more in line with nodes which will be created by node group.
 		mainCreatedNodeInfo, aErr := utils.GetNodeInfoFromTemplate(createNodeGroupResult.MainCreatedNodeGroup, daemonSets, o.taintConfig)
 		if aErr == nil {
-			nodeInfos[createNodeGroupResult.MainCreatedNodeGroup.Id()] = mainCreatedNodeInfo
+			nodeInfos[bestOption.NodeGroup.Id()] = mainCreatedNodeInfo
```

Comment:
Why take NodeGroup from best option, rather than from createNodeGroupResult like before? The new version assumes the group creation mutates the non-existing NG rather than create a new one. Both approaches seem reasonable enough, so why add the assumption?

Reply:
I just added it for the consistency with the `else` statement bellow. We are overriding the bestOption.NodeGroup either way few lines above, and this seems more concise. 

```
bestOption.NodeGroup = createNodeGroupResult.MainCreatedNodeGroup
```

Reply:
I've changed all the `bestOption.NodeGroup` to `createNodeGroupResult.MainCreatedNodeGroup.` throughout the whole "create node group" block.